strax/processing/general.py

```diff
@@ -4,26 +4,48 @@
 import numba
 from numba.typed import List
 import numpy as np
+import warnings
 
 export, __all__ = strax.exporter()
 
 
-# (5-10x) faster than np.sort(order=...), as np.sort looks at all fields
-# TODO: maybe this should be a factory?
 @export
-@numba.jit(nopython=True, nogil=True, cache=True)
 def sort_by_time(x):
-    """Sort pulses by time, then channel.
-
-    Assumes you have no more than 10k channels, and records don't span
-    more than 100 days. TODO: FIX this
+    """Sorts things. Either by time or by time, then channel if both
+    fields are in the given array.
     """
     if len(x) == 0:
-        # Nothing to do, and .min() on empty array doesn't work, so:
         return x
+    
+    if 'channel' in x.dtype.names:
+        channel = x['channel']
+    else:
+        channel = np.ones(len(x))
+        
+    max_time_difference = (np.iinfo(np.int64).max - 10) / np.abs(channel.max()+1)
+    # Subtract 10 to have some extra margin, just in case.
+    # Use absolute to account for peaks which are channel -1.
+    _time_range_too_large = (x['time'].max() - x['time'].min()) > max_time_difference
+    if not _time_range_too_large:
+        # Faster sorting:
+        x = _sort_by_time_and_channel(x, channel, channel.max()+1)
```

Comment:
```suggestion
    max_time_difference = (np.iinfo(np.int64).max - 10) / np.abs(channel.max()+1)
    # Subtract 10 to have some extra margin, just in case.
    # Use absolute to account for peaks which are channel -1.
    _time_range_too_large = (x['time'].max() - x['time'].min()) > max_time_difference
    if not _time_range_too_large:
        # Faster sorting:
        if 'channel' in x.dtype.names:
            channel = x['channel']
        else:
            channel = np.ones(len(x))

        x = _sort_by_time_and_channel(x, channel, channel.max()+1)
```

Reply:
I cannot because I need a channel already in the line before `max_time_difference = (np.iinfo(np.int64).max - 10) / np.abs(channel.max()+1)`.

Reply:
ah yeah I see 👍